packages/eyes/src/generate-snapshots/generate-snapshots.ts

```diff
@@ -0,0 +1,108 @@
+import * as path from 'path';
+const StylableWebpackPlugin = require('@stylable/webpack-plugin');
+const config = require(path.join(process.cwd(), './.autotools/webpack.config.js'));
+import * as webpack from 'webpack';
+import * as glob from 'glob';
+import {HTMLSnapshotPlugin} from '@stylable/webpack-extensions';
+import {createElement} from 'react';
+import {renderToStaticMarkup} from 'react-dom/server';
+import {generateFilteringLogic} from './filter-logic';
+import {generateMapping} from './generate-mapping';
+import Registry, {importMeta, getCompName, IComponentMetadata} from '@ui-autotools/registry';
+import {consoleLog} from '@ui-autotools/utils';
+import {dedent} from './dedent';
+import {parseSnapshotFilename} from './filename-utils';
+
+importMeta();
+
+const mapping = generateMapping(Registry);
+const filteringLogic = generateFilteringLogic(mapping);
+
+const fileNameRegex = /(?:.\/.autotools\/)(.+)/; // Match the file name
+
+async function buildSingleFile(file: string, directory: string) {
+  const entryName = fileNameRegex.exec(file)![1];
+  const snapshotConfig = {
+    entry: {
+      [entryName]: file
+    },
+    target: 'node',
+    output: {
+      filename: '[name].js',
+      pathinfo: true,
+      path: directory
+    },
+    plugins: [
+      new StylableWebpackPlugin({
+        outputCSS: true,
+        path: directory,
+        filename: '[name].css'
+      }),
+      new HTMLSnapshotPlugin({
+        render(compiledFile, sourceFile) {
```

Comment:
Define this as a top-level function to reduce indentation and to make it easier to see which parts of webpack config we adjust.